### PR TITLE
feat(Prerender webpages): Support for prerender.io

### PR DESCRIPTION
**Pull request by Deokumarjnu**
https://github.com/quintype/ace-planning/issues/55

### CODE REVIEW

File: server/create-app.js [hunk 1]
```diff
@@ -47,6 +48,17 @@ function createApp({
       maxAge: "1h",
     })
   );
+  prerenderServiceUrl &&
+    app.use(function (req, res, next) {
```
Comment (sharangj) [inline server/create-app.js:52]: Please move this to isomorphocHandler so that we can add cache headers

File: test/__mocks__/prerender.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+const assert = require("assert").strict;
+const express = require("express");
+
+const supertest = require("supertest");
+const createApp = require("../../server/create-app");
+
+describe("Prerender", function () {
```
Comment (sharangj) [inline test/__mocks__/prerender.js:7]: Name the test file properly. Follow the same conventions

File: server/routes.js [hunk 3]
```diff
@@ -320,6 +321,25 @@ exports.isomorphicRoutes = function isomorphicRoutes(
...
+    app.use((req, res, next) => {
+      if (req.query.preload) {
+        // eslint-disable-next-line global-require
+        require("prerender-node").set(
+          "prerenderServiceUrl",
+          prerenderServiceUrl
+        );
+        res.setHeader(
+          "Cache-Control",
+          "public,max-age=15,s-maxage=60, stale-while-revalidate=150,stale-if-error=3600"
```
Comment (sharangj) [inline server/routes.js:334]: Why not a longer cache header?

File: server/routes.js [hunk 4]
```diff
@@ -303,6 +303,7 @@ exports.isomorphicRoutes = function isomorphicRoutes(
     renderServiceWorker = renderServiceWorkerFn,
     publisherConfig = require("./publisher-config"),
     redirectUrls = [],
+    prerenderServiceUrl = "",
```
Comment (sharangj) [inline server/routes.js:306]: Update documentation

File: server/routes.js (hunk 3 above)
Comment (sharangj) [reply server/routes.js:334]: We need to add cache tags

File: server/routes.js (hunk 4 above)
Comment (Deokumarjnu) [reply server/routes.js:307]: done

**Comment by Deokumarjnu** [comment]
Closing this as we have duplicate  pr [here](https://github.com/quintype/quintype-node-framework/pull/179)

**Pull request closed without merging**